ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -31,73 +31,48 @@ let tdb_tool = !Xapi_globs.tdb_tool
 
 let debug_level = !Xapi_globs.winbind_debug_level |> string_of_int
 
-module Winbind = struct
-  let name = "winbind"
-
-  let is_ad_enabled ~__context =
-    ( Helpers.get_localhost ~__context |> fun self ->
-      Db.Host.get_external_auth_type ~__context ~self )
-    |> fun x -> x = Xapi_globs.auth_type_AD
-
-  let start ~timeout ~wait_until_success =
-    Xapi_systemctl.start ~timeout ~wait_until_success name
-
-  let stop ~timeout ~wait_until_success =
-    Xapi_systemctl.stop ~timeout ~wait_until_success name
-
-  let init_service ~__context =
-    if is_ad_enabled ~__context then
-      start ~wait_until_success:false ~timeout:5.
-    else
-      debug "Skip starting %s as AD is not enabled" name
-
-  let check_ready_to_serve ~timeout =
-    (* we _need_ to use a username contained in our domain, otherwise the following tests won't work.
-       Microsoft KB/Q243330 article provides the KRBTGT account as a well-known built-in SID in AD
-       Microsoft KB/Q229909 article says that KRBTGT account cannot be renamed or enabled, making
-       it the perfect target for such a test using a username (Administrator account can be renamed) *)
-    let resolve_KRBTGT () =
-      try
-        Helpers.call_script ~log_output:Never wb_cmd ["-n"; "KRBTGT"] |> ignore ;
-        true
-      with _ -> false
-    in
-    try
-      Helpers.retry_until_timeout ~timeout
-        (Printf.sprintf "Checking %s ready to serve" name)
-        resolve_KRBTGT ;
-      debug "Service %s is ready to serve request" name
-    with e ->
-      let msg =
-        Printf.sprintf
-          "%s cannot serve after checking for %f seconds, error: %s" name
-          timeout
-          (ExnHelper.string_of_exn e)
-      in
-      error "Service not ready error: %s" msg ;
-      raise (Auth_service_error (E_GENERIC, msg))
-end
+type domain_info = {
```

Comment:
`info` is a bit redundant unless there is a conflict.

Reply:
After re-thinking, I prefer `domain_info` here, as `domain` often refer to the domain name.